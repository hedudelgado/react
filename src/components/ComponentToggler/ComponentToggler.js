import React from 'react';
import { object, string, func } from 'prop-types';

const ComponentToggler = ({
  app,
  viewName,
  view: View,
  edit: Edit,
  ...props
}) => ((app.editViewName === viewName) ? (
  <Edit {...props} app={app} />
) : (
  <View {...props} app={app} />
));

ComponentToggler.propTypes = {
  app: object.isRequired,
  viewName: string.isRequired,
  view: func.isRequired,
  edit: func.isRequired,
};

export default ComponentToggler;
