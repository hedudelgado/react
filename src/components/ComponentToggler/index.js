export { default } from './ComponentToggler';

export { default as ComponentTogglerContainer } from './ComponentTogglerContainer';
