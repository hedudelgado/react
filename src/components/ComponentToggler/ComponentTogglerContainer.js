import { connect } from 'react-redux';
import ComponentToggler from './ComponentToggler';
import {
  PROFILE_UPDATED,
  SET_EDIT_VIEW,
  SET_NOTIFICATION_MESSAGE,
  updateUserProfile as updateProfile,
} from '../App/appActions';

const mapStateToProps = ({
  app,
  countries,
  userProfile: { profile },
  dictionary,
  application,
}) => ({
  app,
  countries,
  profile,
  dictionary,
  application,
});

const mapDispatchToProps = dispatch => ({
  setEditView: viewName => dispatch({
    type: SET_EDIT_VIEW,
    editViewName: viewName,
  }),
  setNotificationMessage: messageName => dispatch({
    type: SET_NOTIFICATION_MESSAGE,
    notificationMessage: messageName,
  }),
  updateProfileStore: profile => dispatch({
    type: PROFILE_UPDATED,
    profile,
  }),
  updateProfile: params => dispatch(updateProfile(params)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ComponentToggler);
