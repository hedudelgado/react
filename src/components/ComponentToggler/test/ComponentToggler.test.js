import React from 'react';
import { shallow } from 'enzyme';

import ComponentToggler from '../ComponentToggler';

describe('ComponentToggler Component', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      app: {},
      viewName: 'test',
      edit: () => (<p id="edit">edit</p>),
      view: () => (<p id="view">view</p>),
    };
    wrapper = shallow(<ComponentToggler
      {...props}
    />);
  });

  describe('Renders ComponentToggler Element', () => {
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Toggling', () => {
    it('should show only the view mode when viewName does not match current view in app state', () => {
      expect(wrapper).toContainExactlyOneMatchingElement('view');
      expect(wrapper).not.toContainExactlyOneMatchingElement('edit');
    });

    it('should show only the edit mode when viewName matches current view in app state', () => {
      props.app.editViewName = 'test';
      wrapper = shallow(<ComponentToggler {...props} />);
      expect(wrapper).toContainExactlyOneMatchingElement('edit');
      expect(wrapper).not.toContainExactlyOneMatchingElement('view');
    });
  });
});
