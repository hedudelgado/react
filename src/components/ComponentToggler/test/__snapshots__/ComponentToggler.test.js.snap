// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentToggler Component Renders ComponentToggler Element should match snapshot 1`] = `
<view
  app={Object {}}
/>
`;
